azure/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -70,13 +70,14 @@ resource "azurerm_virtual_machine" "controllers" {
     custom_data    = "${element(data.ct_config.controller-ignitions.*.rendered, count.index)}"
   }
 
-  # Azure mandates setting an ssh_key, even though Ignition custom_data handles it too
+  # Azure mandates setting an ssh_key, provide just a single key as the
+  # others are handled with Ignition custom_data
   os_profile_linux_config {
     disable_password_authentication = true
 
     ssh_keys {
       path     = "/home/core/.ssh/authorized_keys"
-      key_data = "${var.ssh_keys}"
+      key_data = "${element(var.ssh_keys, 0)}"
```

Comment:
I think if you join the keys with newline here, it should work just fine. Also, they should be put in `/home/core/.ssh/authorized_keys.d`, so `update-ssh-keys` does not override them.

Reply:
Ignition keys are also saved to `~/.ssh/authorized_keys`, any reason why these should be placed in a different file?

Reply:
Hmm, I'm not sure then.

Reply:
Nevermind, we cannot put them in another place: 
https://www.terraform.io/docs/providers/azurerm/r/virtual_machine.html#ssh_keys
"NOTE: Due to a limitation in the Azure VM Agent the only allowed path is /home/{username}/.ssh/authorized_keys."